src/App.tsx

```diff
@@ -1,26 +1,307 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+/* eslint-disable max-len */
+
+// #region Components
+
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { MessageError } from './components/MessageError';
+
+// #endregion
+
+// #region allImport
+
+import { Todo } from './types/Todo';
+import { getTodos, postTodos, deleteTodos, patchTodos } from './api/todos';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+
+// #endregion
 
-const USER_ID = 0;
+export type TypeNewTodo = {
+  id: null;
+  userId: number;
+  title: string;
+  completed: boolean;
+};
+
+const USER_ID = 2881;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  // #region AddFocus
+  const inputRef = useRef<HTMLInputElement>(null);
+  const useRefTimer = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const focusForInput = () => {
+    inputRef.current?.focus();
+  };
+  // #endregion
+
+  // #region AddLoader
+  const [LoderId, setLoderId] = useState<number | null>(null);
+
+  const showLoader = (id: number) => {
+    setLoderId(id);
+  };
+  // #endregion
+
+  // #region State
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [theError, setTheError] = useState<string>('');
+  const [inputValue, setInputValue] = useState<string>('');
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [NewTodo, setCreateNewTodo] = useState<TypeNewTodo | null>(null);
+  const [doDisable, setDoDisable] = useState(false);
+
+  // #endregion
+
+  // #region ShowErro
+
+  const errorTimerRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const showError = (message: string) => {
+    setHasError(true);
+    setTheError(message);
+
+    if (errorTimerRef.current) {
+      clearTimeout(errorTimerRef.current);
+    }
+
+    errorTimerRef.current = setTimeout(() => {
+      setHasError(false);
+    }, 3000);
+  };
+
+  // #endregion
+
+  // #region DoFetch
+
+  const doPost = (titleValue: string) => {
+    if (titleValue.trim().length === 0) {
+      showError('Title should not be empty');
+
+      return;
+    }
+
+    const createNewTodo: TypeNewTodo = {
+      id: null,
```

Comment:
```suggestion
      id: 0,
```
better to make consistent types for todo